src/include/storage/storage_utils.h

```diff
@@ -52,7 +52,20 @@ struct PageElementCursor {
 };
 
 struct PageUtils {
-    static uint32_t getNumElementsInAPage(uint32_t elementSize, bool hasNull);
+    static constexpr uint32_t getNumElementsInAPage(uint32_t elementSize, bool hasNull) {
```

Comment:
Why move from cpp to header file? If you want to keep it inside the header file, mark this function as inline.

Reply:
I'd moved it because so it can be constexpr, and constexpr (according to [cppreference](https://en.cppreference.com/w/cpp/language/constexpr)) implies inline (for cases where it's evaluated at runtime presumably).

Frankly, the static_assert I'd used it for is probably not necessary, but I thought it at least made more sense to make it a single static assert than multiple runtime asserts.

Reply:
Got it. Yeah, this makes sense.